src/machine.rs

```diff
@@ -1120,13 +1117,20 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
             None
         };
 
+        let def_id = frame.instance.def_id();
+        let is_user_relevant = (def_id.is_local()
+            || ecx.machine.local_crates.contains(&def_id.krate))
+            && !frame.instance.def.requires_caller_location(ecx.machine.tcx);
```

Comment:
This is the main potential for slowdown, right? We are now computing this on each push, as opposed to only computing it when someone actually needs it.

But I guess this is cheap enough, pushes are rare enough, and it is needed often enough, that that is worth it?

Reply:
Yes, this is the main potential for slowdown with respect to the previous strategy. And yes, I think there are ways to optimize this more, but we spend 5 times as many cycles in the call to `Frame::current_span` (all the rest of the code in this system totals 0.1% of cycles, `Frame::current_span` is 0.5%). So the slowest part of this is code which was already in here, and all data indicates that overall we are faster, so I favor simplicity.